smac/utils/validate.py

```diff
@@ -308,13 +308,14 @@ def validate_epm(self,
 
         # Predict desired runs
         runs, rh_epm = self.get_runs(config_mode, instance_mode, repetitions, runhistory)
-        try:
+        if hasattr(self.scen, "feature_array") and self.scen.feature_array is not None:
             feature_array_size = len(self.scen.cs.get_hyperparameters()) + self.scen.feature_array.shape[1]
-        except AttributeError:
+        else:
             feature_array_size = len(self.scen.cs.get_hyperparameters())
         X_pred = np.empty((len(runs), feature_array_size))
         for idx, run in enumerate(runs):
-            if hasattr(self.scen, "feature_dict") and run.inst != None:
+            if (hasattr(self.scen, "feature_array") and
```

Comment:
This looks quite error-prone. Would it be possible to always assign a `feature_array` to the scenario and only check if it is `None` here?

Reply:
Absolutely. Actually `Scenario` is already doing that and I do not remember any other reason for using `hasattr`, so I must have missed it when writing the original code.